### PR TITLE
Changes required to use dcmstack with Python 3

### PR DESCRIPTION
**Pull request by ghisvail**
This PR aggregates the list of changes I had to make to be able to call `dcmstack` within a Python 3 virtualenv.

### CODE REVIEW

**Comment by mgxd** [comment]
@moloney any chance you could take a look?

File: src/dcmstack/dcmstack_cli.py [hunk 1]
```diff
@@ -250,10 +250,10 @@ def main(argv=sys.argv):
     #Handle each source directory individually
     for src_dir in args.src_dirs:
         if not os.path.isdir(src_dir):
-            print >> sys.stderr, '%s is not a directory, skipping' % src_dir
+            print('%s is not a directory, skipping' % src_dir, file=sys.stderr)
```
Comment (matthew-brett) [inline src/dcmstack/dcmstack_cli.py]: You need to do ``from __future__ import print_function`` in order for this to work on Python 2 as well.

File: src/dcmstack/extract.py [hunk 2]
```diff
@@ -307,8 +307,8 @@ def get_text(byte_str):
                        'OW or OB' : get_text,
                        'OB or OW' : get_text,
                        'UN' : get_text,
-                       'PN' : unicode,
-                       'UI' : unicode,
+                       'PN' : str,
```
Comment (matthew-brett) [inline src/dcmstack/extract.py]: This is a change of behavior on Python 2 - I don't know whether that matters.  @moloney ?  If it does, then you can do a conditional import at the top of the file something like:

```
unicode_str = unicode if sys.version_info[0] < 3 else str
```

File: src/dcmstack/dcmstack_cli.py (hunk 1 above)
Comment (ghisvail) [inline src/dcmstack/dcmstack_cli.py]: Indeed

**Comment by matthew-brett** [comment]
New changes look fine to me.

**Comment by moloney** [comment]
Going through this now... thanks for all the work so far!

**Comment by satra** [comment]
@moloney - can this be merged?

File: src/dcmstack/extract.py [hunk 3]
```diff
@@ -22,6 +22,9 @@
 #long of a decimal string)
 dicom.config.enforce_valid_values = False
 
+# Python 2 / 3 compatibility
+unicode_str = unicode if sys.version_info[0] < 3 else str
```
Comment (mgxd) [inline src/dcmstack/extract.py]: needs `import sys`

**Comment by r-b-g-b** [comment]
I made a py3 compatibility branch myself before I noticed this PR. Two additional things I noticed:

1. I also needed to change all `xrange` to `range`
2. In `dcmstack.axcodes2ornt`, I needed to make `zip('LPA', 'RAS')` into `list(zip('LPI', 'RAS'))`. That change is because python2 `zip` returns a list while python3 `zip` returns an iterable.

I'll try to pull this branch and add the following commits.

**Comment by yarikoptic** [comment]
eh, we need to get this one finalized... just ran into it again in https://github.com/nipy/heudiconv/pull/225 while thinking that we have it solved already

**Comment by yarikoptic** [comment]
FWIW -- looking into it atm

**Pull request merged**